cmd/microservice-sui-worker/main.go

```diff
@@ -0,0 +1,511 @@
+
+	// EnvWorkerMnemonic for the private mnemonic to use to derive the worker keys
+	EnvWorkerMnemonic = `FLU_SUI_WORKER_MNEMONIC`
+
+	// EnvSuiHttpUrl is the url to use to connect to the HTTP sui endpoint
+	EnvSuiHttpUrl = `FLU_SUI_HTTP_URL`
+
+	// EnvPrizePoolVaultId for the Object ID of the prize pool vault
+	EnvPrizePoolVaultId = `FLU_SUI_PRIZE_POOL_VAULT_ID`
+
+	// EnvScallopVersion for the version of the scallop market to use
+	EnvScallopVersion = `FLU_SUI_SCALLOP_VERSION`
+
+	// EnvScallopMarket for the address of the scallop market to use
+	EnvScallopMarket = `FLU_SUI_SCALLOP_MARKET`
+
+	// EnvCoinReserve for the coin reserve containing the fluid token
+	EnvCoinReserve = `FLU_SUI_COIN_RESERVE`
+
+	// SuiClockId for the constant address of the Sui clock
+	SuiClockId = `0x0000000000000000000000000000000000000000000000000000000000000006`
+)
+
+// TODO - doesn't make utility mining payouts
+func main() {
+	var (
+		utilities      = suiApps.UtilityListFromEnvOrFatal(EnvUtilityContracts)
+		baseTokenName  = util.GetEnvOrFatal(EnvUnderlyingTokenName)
+		decimalPlaces_ = util.GetEnvOrFatal(EnvTokenDecimals)
+		suiHttpUrl     = util.PickEnvOrFatal(EnvSuiHttpUrl)
+		workerMnemonic = util.GetEnvOrFatal(EnvWorkerMnemonic)
+
+		underlyingPackageId = util.GetEnvOrFatal(EnvUnderlyingPackageId)
+		fluidPackageId      = util.GetEnvOrFatal(EnvFluidPackageId)
+		coinReserve         = util.GetEnvOrFatal(EnvCoinReserve)
+
+		prizePoolVault = util.GetEnvOrFatal(EnvPrizePoolVaultId)
+		scallopVersion = util.GetEnvOrFatal(EnvScallopVersion)
+		scallopMarket  = util.GetEnvOrFatal(EnvScallopMarket)
+	)
+
+	payoutArgs := payoutArgs{
+		PrizePoolVault: prizePoolVault,
+		ScallopVersion: scallopVersion,
+		ScallopMarket:  scallopMarket,
+		Clock:          SuiClockId,
+	}
+
+	decimalPlaces, err := strconv.Atoi(decimalPlaces_)
+
+	if err != nil {
+		log.Fatal(func(k *log.Log) {
+			k.Format(
+				"Failed to convert the decimals from %v! Was %#v!",
+				decimalPlaces_,
+				decimalPlaces,
+			)
+		})
+	}
+
+	signer, err := signer.NewSignertWithMnemonic(workerMnemonic)
+	if err != nil {
+		log.Fatal(func(k *log.Log) {
+			k.Message = "Failed to create a signer from a mnemnoic string!"
+			k.Payload = err
+		})
+	}
+	workerAddress := signer.Address
+
+	client := suiSdk.NewSuiClient(suiHttpUrl)
+
+	decimalsAdjusted := math.Pow10(decimalPlaces)
+	decimalsRat := new(big.Rat).SetFloat64(decimalsAdjusted)
+
+	fluidToken := sui.SuiToken{
+		TokenShortName: baseTokenName,
+		TokenDecimals:  decimalPlaces,
+		PackageId:      fluidPackageId,
+		IsFluid:        true,
+	}
+
+	baseToken := sui.SuiToken{
+		TokenShortName: baseTokenName,
+		TokenDecimals:  decimalPlaces,
+		PackageId:      underlyingPackageId,
+		IsFluid:        false,
+	}
+
+	sui_queue.DecoratedTransfers(func(decoratedTransfers []sui_queue.DecoratedTransfer) {
+		transfersWithFees := make([]worker_types.TransferWithFee, 0)
+
+		for _, transfer := range decoratedTransfers {
```

Comment:
Explain what gets done during this block in a high level, then connect it to the story